### DISCUSSION TITLE
Disabled Icons

### DISCUSSION THREAD

**Discussion by ramoneguru** [twilio-labs/paste#1115] [General] [answered]
Is there a way to make icons show up as disabled (light mask over them)?

I have icons for capabilities that I'd like to show depending on whether or not it has a certain capability. When it's not capable, I can remove the icon and show a blank space. This is good for fast scanning a list, but it looks a touch odd to me. Alternatively, I can stack the capabilities without any gaps, but it might be tougher to see what is missing.

Ideally, I'd like to have all of the capabilities show up, then disable (using a disabled icon?) the ones that aren't available.

**Icons with gaps**
<img width="1083" alt="Screen Shot 2021-01-27 at 10 33 16 AM" src="https://user-images.githubusercontent.com/476482/106040134-dc258100-608e-11eb-8b0c-dc3ae5a91405.png">

**Icons without any gaps**
<img width="1077" alt="Screen Shot 2021-01-27 at 10 15 46 AM" src="https://user-images.githubusercontent.com/476482/106040130-da5bbd80-608e-11eb-9d27-2fa58ca642fe.png">

**Comment by serifluous** [comment]
Hey @ramoneguru ! Since you're using buttons for these icons, you certainly have the option to use [a disabled button](https://paste.twilio.design/components/button#disabled). However, because a disabled button can't be focused and there's no supporting text here about why the 2 different types of icons are different, it's hard (for us) to know whether it'd be immediately clear to your customers what the UI is trying to tell them about a specific phone number's capabilities.

I'd definitely suggest getting more input from the Product Designers on your team, if you haven't already, to reach a solution or even more understanding of how a customer might approach this screen.

**Comment by ramoneguru** [comment]
Oh, I don't think I'm using it as a button? Were you suggesting wrapping each element in a `<button>` component?

```
<Stack orientation="horizontal">
            <Box width="50px" display="inline-block">
              {item.capabilities.voice.isCapable && (
                <VoiceCapableIcon decorative={false} title="Voice Capable" />
              )}
            </Box>
            <Box width="50px" display="inline-block">
              {item.capabilities.sms.isCapable && (
                <SMSCapableIcon decorative={false} title="SMS Capable" />
              )}
            </Box>
            <Box width="50px" display="inline-block">
              {item.capabilities.mms.isCapable && (
                <MMSCapableIcon decorative={false} title="MMS Capable" />
              )}
            </Box>
            <Box width="50px" display="inline-block">
              {item.capabilities.sip.isCapable && (
                <ProductElasticSIPTrunkingIcon decorative={false} title="SIP Capable" />
              )}
            </Box>
            <Box width="50px" display="inline-block">
              {item.capabilities.fax.isCapable && (
                <FaxCapableIcon decorative={false} title="Fax Capable" />
              )}
            </Box>
</Stack>
```

**Reply by TheSisb** [reply]
Yup, [icons shouldn't be interactive](https://paste.twilio.design/icons/usage-guidelines#icons-shouldnt-be-interactive) for accessibility.  Here's an [example of an icon button](https://paste.twilio.design/components/button#icon-only-button).

**Reply by serifluous** [reply]
Ah yeah, I was mistaken. I was looking at the Active Numbers page.

But @TheSisb 's got the right links ^ and regardless, I'm not 100% sure if disabling a button is the right solution here so definitely rely on your product designer for this!

**Comment by aayushpi** [comment]
Jumping in here, it might be valuable to consider having a table header which shows voice | sms | fax | etc and each column underneath shows either a ☑️  or ❌ - from an information perspective, this will be much clearer. @ramoneguru

**Comment by jweintraub** [comment]
+1 to Aayush's comment. Also, this pattern, if changed, should be changed
everywhere that it appears.

Johanna Weintraub
Staff, Product Designer
[image: twilio] <http://www.twilio.com/?utm_source=email_signature>

**Reply by ramoneguru** [reply]
I guess I'd need to implement a fixed header thing that stops when the user scrolls to the table. Not terribly difficult *shrug*

**Comment by ramoneguru** [comment]
What about this layout? 

Hmmm, looking at it in the list it's still kinda difficult to see what's what. They all kinda run together. I guess if there was an even lighter text color it might work or I could mess around more with the opacity as well...

<img width="826" alt="Screen Shot 2021-01-28 at 4 25 18 PM" src="https://user-images.githubusercontent.com/476482/106215288-75cf5a00-6185-11eb-8a13-a5c0f3fba816.png">

```
<Stack orientation="horizontal" spacing="space0">
            <Box width="40px" display="inline-block">
              {item.capabilities.voice.isCapable
                ? <VoiceCapableIcon decorative={false} title="Voice Capable" color="colorTextLinkDarker" />
                : <VoiceCapableIcon decorative={false} title="Not Voice Capable" color="colorTextWeaker" />
              }
            </Box>
            <Box width="40px" display="inline-block">
              {item.capabilities.sms.isCapable
                ? <SMSCapableIcon decorative={false} title="SMS Capable" color="colorTextLinkDarker" />
                : <SMSCapableIcon decorative={false} title="Not SMS Capable" color="colorTextWeaker" />
              }
            </Box>
            <Box width="40px" display="inline-block">
              {item.capabilities.mms.isCapable
                ? <MMSCapableIcon decorative={false} title="MMS Capable" color="colorTextLinkDarker" />
                : <MMSCapableIcon decorative={false} title="Not MMS Capable" color="colorTextWeaker" />
              }
            </Box>
            <Box width="40px" display="inline-block">
              {item.capabilities.sip.isCapable
                ? <ProductElasticSIPTrunkingIcon decorative={false} title="SIP Capable" color="colorTextLinkDarker" />
                : <ProductElasticSIPTrunkingIcon decorative={false} title="Not SIP Capable" color="colorTextWeaker" />
              }
            </Box>
            <Box width="40px" display="inline-block">
              {item.capabilities.fax.isCapable
                ? <FaxCapableIcon decorative={false} title="Fax Capable" color="colorTextLinkDarker" />
                : <FaxCapableIcon decorative={false} title="Not Fax Capable" color="colorTextWeaker"/>
              }
            </Box>
</Stack>

```

**Reply by aayushpi** [reply]
My concern with this is it doesn't immediately indicate the you _cannot ever_ use phone or sms if it's disabled. It can also be considered as "temporarily not available" and require a user to hover to learn more. A clear check or no check is a lot more effective in making the case. I am also unsure you'll need a fixed header since you can still pass a title attribute saying "SMS not available" on hover for the icon itself. @ramoneguru

**Comment by jweintraub** [comment, marked as answer]
Please check with Luz @lgalloherran before
changing it - it would be good for her to approve the design changes. I
think Aayush's idea made sense.

Johanna Weintraub
Staff, Product Designer
[image: twilio] <http://www.twilio.com/?utm_source=email_signature>

**Reply by corinne-nagel** [reply]
@ramoneguru Here's a design idea we worked on a bit ago for easy scanning (hasn't been tested but wanted to share with you all/Luz as an idea)
![Screen Shot 2021-01-29 at 9 28 03 AM](https://user-images.githubusercontent.com/62260820/106301177-a3171900-6214-11eb-929b-133d5c003f77.png)

**Comment by ramoneguru** [comment]
Nice, yeah will discuss with Luz!!! Thanks for the feedback!!!
